src/robot.cpp

```diff
@@ -33,17 +38,11 @@ auto Robot::from(std::shared_ptr<moveit::core::RobotModel const> const& model,
 
         auto var = Variable{};
 
-        bool bounded = bounds.position_bounded_;
-
+        var.bounded = bounds.position_bounded_;
         var.min = bounds.min_position_;
         var.max = bounds.max_position_;
 
-        var.clip_min = bounded ? var.min : std::numeric_limits<double>::lowest();
-        var.clip_max = bounded ? var.max : std::numeric_limits<double>::max();
-
-        var.span = var.max - var.min;
-
-        if (!(var.span >= 0 && var.span < std::numeric_limits<double>::max())) var.span = 1;
+        var.span = var.bounded ? var.max - var.min : kUnboundedVariableSpan;
```

Comment:
Should `var.span` just be a function?

Reply:
I think we want this to be a variable so it doesn't have to be needless recalculated multiple times... but with your next comment, I like the idea of providing abstractions to clip variables to limits / generate random values, and then no other code besides these functions would touch the span member variable.